cider/exec/template/CastIR.cpp

```diff
@@ -58,21 +58,24 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCastFun(
   CHECK_EQ(uoper->get_optype(), kCAST);
   auto ti = uoper->get_type_info();
   auto operand = uoper->get_operand();
-  const auto operand_as_const = dynamic_cast<const Analyzer::Constant*>(operand);
-
-  auto operand_lv = codegen(operand, co, true);
-
-  // TODO: Refactor codegenCast.
   auto operand_ti = operand->get_type_info();
-  operand_ti.set_notnull(true);
-  ti.set_notnull(true);
 
-  auto fixedsize_lv = dynamic_cast<FixedSizeColValues*>(operand_lv.get());
-  CHECK(fixedsize_lv);
-  auto cast_lv =
-      codegenCast(fixedsize_lv->getValue(), operand_ti, ti, operand_as_const, co);
+  auto operand_nullable = codegen(operand, co, true);
+  auto operand_fixedsize = dynamic_cast<FixedSizeColValues*>(operand_nullable.get());
+  CHECK(operand_fixedsize);
+  auto operand_lv = operand_fixedsize->getValue();
+  llvm::Value* target_lv = nullptr;
 
-  return std::make_unique<FixedSizeColValues>(cast_lv, fixedsize_lv->getNull());
+  if (operand_lv->getType()->isIntegerTy()) {
+    if (ti.is_integer() || ti.is_decimal() || ti.is_boolean()) {
+      target_lv = codegenCastBetweenIntTypesForArrow(operand_lv, operand_ti, ti);
```

Comment:
will (cast int32 to int64) fall into this branch?

Reply:
`is_integer()` including tinyint,smallint,int and bigint.